code/modules/mob/living/death.dm

```diff
@@ -91,6 +91,14 @@
 	GLOB.alive_mob_list -= src
 	GLOB.dead_mob_list += src
 	if(mind)
+		if(mind.name && !isbrain(src)) // !isbrain() is to stop it from being called twice
+			var/turf/T = get_turf(src)
+			var/area_name = get_area_name(T)
+			for(var/P in GLOB.player_list)
```

Comment:
`GLOB.dead_mob_list` may work here and will be a smaller list to iterate over.

Reply:
That will also include dead monkeys. Dead simple mobs and what not. You then also need to check if the mob has a client which is "free" in this list

Reply:
If it counts all dead mobs it'll probably be better to just keep using `GLOB.player_list`.

Reply:
All observers are put into that list, and its used in other places too.
```dm
		// Show it to ghosts
		for(var/mob/M in GLOB.dead_mob_list)
			if(isobserver(M) && M.client && (M.client.prefs.toggles & PREFTOGGLE_CHAT_GHOSTPDA))
				var/ghost_message = "<span class='name'>[pda.owner]</span> ([ghost_follow_link(pda, ghost=M)]) <span class='game say'>PDA Message</span> --> <span class='name'>[P.owner]</span> ([ghost_follow_link(P, ghost=M)]): <span class='message'>[t]</span>"
				to_chat(M, "[ghost_message]")
```
Please change it